src/utils/components/DiskSourceFlyoutMenu/DiskSourceFlyoutMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC, useRef, useState } from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  Menu,
+  MenuContainer,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  MenuToggle,
+} from '@patternfly/react-core';
+
+import { getDiskSourceOptionGroups } from '../DiskModal/components/DiskSourceSelect/utils/utils';
+import { SourceTypes } from '../DiskModal/utils/types';
+
+import FlyoutItem from './FlyoutItem/FlyoutItem';
+
+export type DiskSourceFlyoutMenuProps = {
+  className?: string;
+  isTemplate?: boolean;
+  onSelect: (value: SourceTypes) => void;
+};
+
+const DiskSourceFlyoutMenu: FC<DiskSourceFlyoutMenuProps> = ({
+  className,
+  isTemplate = false,
+  onSelect,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const menuRef = useRef<HTMLDivElement>(null);
+  const toggleRef = useRef<HTMLButtonElement>(null);
+
+  const options = getDiskSourceOptionGroups(isTemplate);
+
+  const onToggleClick = () => {
+    setIsOpen((open) => !open);
+  };
+
+  const onSelectSource = (value: SourceTypes) => {
+    onSelect(value);
+    setIsOpen(false);
+  };
+
+  const toggle = (
+    <MenuToggle
+      className={className}
+      isExpanded={isOpen}
+      onClick={onToggleClick}
+      ref={toggleRef}
+      variant="primary"
+    >
+      {t('Add disk')}
+    </MenuToggle>
+  );
+
+  const menu = (
+    <Menu containsFlyout ref={menuRef}>
+      <MenuContent>
+        <MenuList>
+          {options?.map(({ description, groupLabel, items }) => {
+            const isFlyout = items?.length > 1;
```

Comment:
no need its only used once

Reply:
I can do this, I thought it would be more readable like this when the condition has context to it